### PR TITLE
Ability to access the current task object from within the task function

### PR DESCRIPTION
**Pull request by thomasst**
Ability to access the current task object from within the task function using `tiger.current_task` (or `tiger.current_tasks` in case of a batch function).

### CODE REVIEW

**Comment by wojcikstefan** [comment]
LGTM, though I think the tests' structure is pretty confusing - it's not immediately clear that `current_task` and `current_tasks` come from `tests/tasks.py` (don't see any explicit imports).

File: tasktiger/_internal.py [hunk 1]
```diff
@@ -15,6 +15,14 @@
 SCHEDULED = 'scheduled'
 ERROR = 'error'
 
+# Global task context. We store this globally (and not on the TaskTiger
+# instance) for consistent results just in case the user has multiple TaskTiger
```
Comment (wojcikstefan) [inline tasktiger/_internal.py]: In case of multiple intances, is there a way to know which task in `current_tasks` belongs to which TaskTiger instance?

Comment (thomasst) [inline tasktiger/_internal.py]: It's irrelevant, since `current_task` within a task would always refer to the currently processed task.

**Comment by anemitz** [comment]
@thomasst bump

**Pull request merged**